### DISCUSSION TITLE
Serving SPA/React App?

### DISCUSSION THREAD

**Discussion by apikee** [pocketbase/pocketbase#879] [Q&A]
Hello, i'm using Pocketbase as a Framework and I would like to serve React app with router. I'am able to serve default page, but any other router related routes are handled by Pocketbase instead of React Router. I understand why this is happening, would know how to solve it in Go Fiber, but I'am unable to solve it in Pocketbase. Can someone help? The current code is:
```
e.Router.GET("/*", echo.StaticDirectoryHandler(os.DirFS("public"), false))
```
What needs to be added?

In Go Fiber, i would do this:
```
func (t *Transport) setupStaticServing() {
	t.App.Static("/", "public")
	t.App.Static("*", "public/index.html")
}
```
`App` being Fiber App.

Thank you

**Comment by ganigeorgiev** [comment]
You have to use `apis.StaticDirectoryHandler`, not `echo.StaticDirectoryHandler` because of a conflict with the `RemoveTrailingSlashMiddleware`.

Additioanlly, PocketBase v0.8.0-rc1 comes with new `StaticDirectoryHandler` where you can allow index.html forwarding on missing files by changing the last argument to `true`:

```go
package main

import (
    "log"
    "os"

    "github.com/pocketbase/pocketbase"
    "github.com/pocketbase/pocketbase/apis"
    "github.com/pocketbase/pocketbase/core"
)

func main() {
    app := pocketbase.New()

    app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
        // serves static files from the provided public dir (if exists)
        e.Router.GET("/*", apis.StaticDirectoryHandler(os.DirFS("public"), true))

        return nil
    })

    if err := app.Start(); err != nil {
        log.Fatal(err)
    }
}
```

_Note1: Notice that we are using `apis.StaticDirectoryHandler`, not the default `echo.StaticDirectoryHandler`._
_Note2: The index.html fallback forwarding works only with PocketBae v0.8.0-rc1+_

**Reply by apikee** [reply]
Thanks for the answer. I see v0.8.0-rc1 should be released in few weeks, so i'll wait.

**Reply by ganigeorgiev** [reply]
I don't plan making changes to this part of the code, so you can give it a try by updating the dependency in your `go.mod` file to:

```
github.com/pocketbase/pocketbase v0.8.0-rc1
```
And then run `go get` (or `go mod tidy`).

**Reply by apikee** [reply]
yeah that seems to work, but have tested it on separate clean project, since the one im currently working on have several parts that would break. I'm quite interested to see what else will change until v1.0.0.
